src/daemon/bundle.rs

```diff
@@ -87,6 +89,7 @@ pub async fn load_actor_bundles_from_server(
     network: &NetworkChain,
     bundles: &[ActorBundleInfo],
 ) -> anyhow::Result<Vec<Cid>> {
+    let semaphore = Arc::new(Semaphore::new(4));
```

Comment:
Why 4 in particular? And not 3 or 10?

Reply:
I think any value between 2-8 is fine, 4 is somewhere in the middle.

Reply:
Larger number is faster but more error-prone. Previously, all tasks run in parallel and the test was quite flaky

Reply:
Looking at the time cost, 4 makes it more stable and still fast

Reply:
We can try, but I'm not :100: convinced. I'd rather we have a deterministically correct and working solution. This seems like reducing error probability from an unknown chance to another unknown (but likely smaller) chance, while always paying the price of reduced performance (slower startup for fresh nodes).

Reply:
We've been using the same logic for RPC snapshot test and it's much more stable, the only notable difference is, actor bundle uses Github assets primarily while RPC snapshot test uses DO space CDN.

Reply:
And I believe it's a common practice to limit max concurrent downloads, e.g. the author uses 8 in https://patshaughnessy.net/2020/1/20/downloading-100000-files-using-async-rust